cmake/check_flags.cmake

```diff
@@ -0,0 +1,7 @@
+include (CheckCXXCompilerFlag)
+include (CheckCCompilerFlag)
+
+check_cxx_compiler_flag("-Wsuggest-destructor-override" HAS_SUGGEST_DESTRUCTOR_OVERRIDE)
+check_cxx_compiler_flag("-Wshadow" HAS_SHADOW)
+check_cxx_compiler_flag("-Wsuggest-override" HAS_SUGGEST_OVERRIDE)
+check_cxx_compiler_flag("-Wsuggest-destructor-override" HAS_SUGGEST_DESTRUCTOR_OVERRIDE)
```

Comment:
The same flag `-Wsuggest-destructor-override` checked twice?

Reply:
Oops. Removed.